docs/native-modules-ios.md

```diff
@@ -1,442 +1,466 @@
+
+### Test What You Have Built
+
+At this point you have set up the basic scaffolding for your native module in iOS. Test that out by accessing the native module and invoking it’s exported method in JavaScript.
+
+Find a place in your application where you would like to add a call to the native module’s `createCalendarEvent` method. Below is an example of a component, `NewModuleButton` you can add in your app. You can invoke the native module inside `NewModuleButton`'s `onPress` function.
 
 ```jsx
-CalendarManager.addEvent(
-  'Birthday Party',
-  '4 Privet Drive, Surrey',
-  date.getTime()
-); // passing date as number of milliseconds since Unix epoch
+import React from 'react';
+import { NativeModules, Button } from 'react-native';
+
+const NewModuleButton = () => {
+  const onPress = () => {
+    console.log('We will invoke the native module here!');
+  };
+
+  return (
+    <Button
+      title="Click to invoke your native module!"
+      color="#841584"
+      onPress={onPress}
+    />
+  );
+};
+
+export default NewModuleButton;
 ```
 
-or
+In order to access your native module from Javascript you need to first import `NativeNodules` from React Native:
 
 ```jsx
-CalendarManager.addEvent(
-  'Birthday Party',
-  '4 Privet Drive, Surrey',
-  date.toISOString()
-); // passing date as ISO-8601 string
+import { NativeModules } from 'react-native';
 ```
 
-And both values would get converted correctly to the native `NSDate`. A bad value, like an `Array`, would generate a helpful "RedBox" error message.
+You can then access the `CalendarModule` native module off of `NativeModules`.
 
-As `CalendarManager.addEvent` method gets more and more complex, the number of arguments will grow. Some of them might be optional. In this case it's worth considering changing the API a little bit to accept a dictionary of event attributes, like this:
+```jsx
+const { CalendarModule } = NativeModules;
+```
 
-```objectivec
-#import <React/RCTConvert.h>
+Now that you have the CalendarModule native module available, you can invoke your native method `createCalendarEvent`. Below it is added to the `onPress` method in `NewModuleButton`:
 
-RCT_EXPORT_METHOD(addEvent:(NSString *)name details:(NSDictionary *)details)
-{
-  NSString *location = [RCTConvert NSString:details[@"location"]];
-  NSDate *time = [RCTConvert NSDate:details[@"time"]];
-  ...
-}
+```jsx
+const onPress = () => {
+  CalendarModule.createCalendarEvent('testName', 'testLocation');
+};
 ```
 
-and call it from JavaScript:
+The final step is to rebuild the React Native app so that you can have the latest native code (with your new native module!) available. In your command line, where the react native application is located, run the following :
 
 ```jsx
-CalendarManager.addEvent('Birthday Party', {
-  location: '4 Privet Drive, Surrey',
-  time: date.getTime(),
-  description: '...'
-});
+react-native run-ios
```

Comment:
We should not recommend using global CLI, please convert this line to `npx react-native run-ios` and change the code block language from `jsx` to `shell`.

Reply:
done